streams/src/test/java/org/apache/kafka/streams/tests/EosTestDriver.java

```diff
@@ -383,7 +403,84 @@ private static void verifySum(final Map<TopicPartition, List<ConsumerRecord<byte
                     currentSumPerKey.put(key, sum);
 
-                    if (!receivedKey.equals(key) || receivedValue != sum) {
+                    if (!receivedKey.equals(key) || receivedValue != sum.longValue()) {
+                        throw new RuntimeException("Result verification failed for " + receivedRecord + " expected <" + key + "," + value + "> but was <" + receivedKey + "," + receivedValue + ">");
+                    }
+                }
+            } catch (final NullPointerException e) {
+                System.err.println(inputPerTopicPerPartition);
+                e.printStackTrace(System.err);
+                throw e;
+            }
+        }
+    }
+
+    private static void verifyMax(final Map<TopicPartition, List<ConsumerRecord<byte[], byte[]>>> inputPerTopicPerPartition,
+                                  final Map<TopicPartition, List<ConsumerRecord<byte[], byte[]>>> maxPerTopicPerPartition) {
+        final StringDeserializer stringDeserializer = new StringDeserializer();
+        final IntegerDeserializer integerDeserializer = new IntegerDeserializer();
+
+        final HashMap<String, Integer> currentMinPerKey = new HashMap<>();
+        for (final Map.Entry<TopicPartition, List<ConsumerRecord<byte[], byte[]>>> partitionRecords : maxPerTopicPerPartition.entrySet()) {
+            try {
+                final TopicPartition inputTopicPartition = new TopicPartition("repartition", partitionRecords.getKey().partition());
+                final Iterator<ConsumerRecord<byte[], byte[]>> inputRecords = inputPerTopicPerPartition.get(inputTopicPartition).iterator();
+
+                for (final ConsumerRecord<byte[], byte[]> receivedRecord : partitionRecords.getValue()) {
+                    final ConsumerRecord<byte[], byte[]> input = inputRecords.next();
+
+                    final String receivedKey = stringDeserializer.deserialize(receivedRecord.topic(), receivedRecord.key());
+                    final int receivedValue = integerDeserializer.deserialize(receivedRecord.topic(), receivedRecord.value());
+                    final String key = stringDeserializer.deserialize(input.topic(), input.key());
+                    final int value = integerDeserializer.deserialize(input.topic(), input.value());
+
+
+                    Integer max = currentMinPerKey.get(key);
+                    if (max == null) {
+                        max = value;
+                    }
+                    max = Math.max(max, value);
+                    currentMinPerKey.put(key, max);
+
+                    if (!receivedKey.equals(key) || receivedValue != max.intValue()) {
+                        throw new RuntimeException("Result verification failed for " + receivedRecord + " expected <" + key + "," + value + "> but was <" + receivedKey + "," + receivedValue + ">");
+                    }
+                }
+            } catch (final NullPointerException e) {
+                System.err.println(inputPerTopicPerPartition);
```

Comment:
Just thinking about this twice, a NPE should not happen here any more as we reworked the code. IMHO, we can just remove the whole try-catch-block. WDYT?